rfcs/20201014-VTune-ITT-tagging/README.md

```diff
@@ -0,0 +1,142 @@
+For example, users can understand one inner primitive contains several gemm_nocopy_f32 kernels.  
+![](images/ops_mapping.JPG "mapping between oneDNN primitives and kernels")
+
+## Proposal
+
+Add VTune ITT tasks for all primitives.
+Users can see profiling results grouped by primitives. 
+primitive grouping is more useful than function grouping to DL developers.
+### Implementation
+To enable VTune ITT tagging, we includes ittnotify related source files in projects https://github.com/intel/ittapi under src/common/ittnofiy folder.  
+We also create ittnotify.cpp and ittnotify.hpp under /src/common folder to use ITT functions from copied ITT source files.
+We introduce a new namespace itt under dnnl namespace.
+In the same sapce we have 4 DNNL_API for using ITT tagging.
+
+| functions  | Note |  
+|---------------------|---------------------|  
+| void DNNL_API primitive_task_start(primitive_kind_t kind) |  use this function to tag start of a ITT task with primitive kind info  | 
+| void DNNL_API primitive_task_end()  |    use this function to tag end of a ITT task    | 
+| primitive_kind_t DNNL_API primitive_task_get_current_kind() |   get the primitive task for current ITT task       | 
+| bool DNNL_API get_itt(__itt_task_level level) |    get ITT level.  enable different ITT tagging by using this level variable  |   
+
+We also define a enum type with different itt task level.  
+Here is a table for meaning of all itt task levels.  
+
+| __itt_task_level  | Note |  
+|---------------------|---------------------|  
+| __itt_task_level_none |   no ITT event will be triggered  | 
+| __itt_task_level_low  |    ITT events are only triggered in master thread     | 
+| __itt_task_level_high |    ITT events are triggered in all OMP/TBB threads      |   
+
+There is a new struct itt_task_level_t with only one int variable in it.  
+~~~cpp
+struct itt_task_level_t {
+    int level;
+};
+~~~
+
+#### Implementation details :  tag primitive execution with ITT
+---
+Here are pesudo codes for how we define primitive execution on master thread as a ITT task.
+~~~cpp
+status_t dnnl_primitive_execute(const primitive_iface_t *primitive_iface,
+    ....
+    if (itt::get_itt(itt::__itt_task_level_low))
+        itt::primitive_task_start(
+            primitive_iface->pd()->impl()->kind());
+
+    .... // Primitive Execution
+
+    if (itt::get_itt(itt::__itt_task_level_low))
+        itt::primitive_task_end();
+    return status;
+}
+
+~~~
+
+### Feature Enablement
+---
+**Compilation Time**  
+
+We enable this feature on compilation time by default.  
+User can disable the feature by following CMake declaration at compiling time.    
+DNNL_ITT_TASKS=False   
+Moreover, user can give a custom VTune path if they don't have it under /opt/intel/...   
+CMAKE_VTUNE_HOME=<CUSTOM PATH>  
+
+**Runtime**  
+
+Users can use a environment variable to control ITT level at runtime.  
+
+| DNN_ITT   | __itt_task_level  | Note |  
```

Comment:
I think the variable should be more aligned with vtune. Perhaps we can have `DNNL_VTUNE_TAGS_LEVEL`? tagging @emfomenk  (no pun intended)

Reply:
I changed it accordingly, but I use DNNL_VTUNE_TASK_LEVEL instead.
if it works for all, I will also make the changes in MR.
https://gitlab.devtools.intel.com/ipl/dl/mkl-dnn/-/merge_requests/2091

Reply:
Given the feedback from VTune team, I suppose this should now be `DNNL_ITT_TASK_LEVEL`.